bika/lims/browser/analysisrequest/publish.py

```diff
@@ -1404,6 +1430,22 @@ def _analysis_data(self, analysis, decimalmark=None):
                 if ret and ret['out_of_range']:
                     andict['outofrange'] = True
                     break
+
+        # Get unit conversions for result
+        st_uid = None
+        if sample and sample.get('sample_type') and \
+           sample['sample_type'].get('obj'):
+            st_uid = sample['sample_type']['obj'].UID()
+        if st_uid:
+            for unit_conversion in analysis.getUnitConversions():
+                if unit_conversion.get('SampleType') and \
+                   unit_conversion.get('Unit') and \
+                   unit_conversion.get('SampleType') == st_uid:
+                    andict['unit_conversions'].append(unit_conversion['Unit'])
+                    if andict['hide_primary_result'] == False:
+                        andict['hide_primary_result'] = \
+                            unit_conversion.get('HidePrimaryUnit', '0') == '1'
```

Comment:
Please refactor that, it is not understandable and it looks like a huge regular expression. Sorry for being harsh, but this is not maintainable.

Reply:
Refactored and added comments